contrib/golang/filters/http/source/go/pkg/http/capi.go

```diff
@@ -24,6 +24,7 @@ import (
 )
 
 type HttpCAPI interface {
+	//TODO: we should switch all the request types to be using `r *httpRequest` for consistency.
```

Comment:
we'd better use unsafe.Pointer for them, and convert unsafe.Pointer to httpRequest in the implementation.
here is the example:
https://github.com/envoyproxy/envoy/pull/26173/files#diff-d0977c943942fbbf3c44b7e48ce284c00116a0b6b66cc3c3a26687959d31d84aR220-R221

Reply:
done, thanks.